core/storage/base_model/gae_models.py

```diff
@@ -720,6 +720,73 @@ def delete(self, committer_id, commit_message, force_deletion=False):
                 committer_id, self._COMMIT_TYPE_DELETE, commit_message,
                 commit_cmds)
 
+    @classmethod
+    def delete_multi(cls, entity_ids, committer_id,
+                     commit_message, force_deletion=False):
+        """Deletes the given cls instancies with the given entity_ids.
+
+        Args:
+            entity_ids: list(str). Ids of entities to delete.
+            committer_id: str. The user_id of the user who committed the change.
+            commit_message: str.
+            force_deletion: bool. If True these models are deleted completely
+                from storage, otherwise there are only marked as deleted.
+                Default is False.
+
+        Raises:
+            Exception: This model instance has been already deleted.
+        """
+        versioned_models = cls.get_multi(entity_ids)
+        if force_deletion:
+            all_models_metadata_keys = []
+            all_models_content_keys = []
+            for model in versioned_models:
+                model_version_numbers = [
+                    python_utils.UNICODE(num + 1) for num in
+                    python_utils.RANGE(model.version)]
+                model_snapshot_ids = [
+                    model.get_snapshot_id(model.id, version_number)
+                    for version_number in model_version_numbers]
+
+                all_models_metadata_keys.extend([
+                    ndb.Key(model.SNAPSHOT_METADATA_CLASS, snapshot_id)
+                    for snapshot_id in model_snapshot_ids])
+                all_models_content_keys.extend([
+                    ndb.Key(model.SNAPSHOT_CONTENT_CLASS, snapshot_id)
+                    for snapshot_id in model_snapshot_ids])
+
+            ndb.delete_multi(
```

Comment:
Should this run in a single transaction together with the deletion of the versioned models?

Reply:
Done.